.orchestra/ci/ci-hook/app.py

```diff
@@ -89,77 +159,309 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         assert not self.user.impersonationtokens.get(impersonation_token_id).active
 
 
-def trigger_ci(username, repo_url, ref, before, after):
+def impersonate(admin_gl, username):
+    matching = admin_gl.users.list(username=username)
+    if len(matching) != 1:
+        raise Exception("Unexpected number of users matching " + username)
+    user_id = matching[0].id
+
+    user = admin_gl.users.get(user_id)
+    return ImpersonationToken(user)
+
+
+def trigger_ci(username, repo_url, base_repo_url, ref, status_update_metadata: Optional[dict] = None):
     # Ignore pushes by the CI itself
     if username == ci_user:
         return
 
     admin_gl = gitlab.Gitlab(GITLAB_URL, private_token=ADMIN_TOKEN)
     admin_gl.auth()
 
-    matching = admin_gl.users.list(username=username)
-    if len(matching) != 1:
-        raise Exception("Unexpected number of users matching " + username)
-    user_id = matching[0].id
+    try:
+        with impersonate(admin_gl, username) as token:
+            user_gl = gitlab.Gitlab(GITLAB_URL, private_token=token)
+            user_gl.auth()
+            project = user_gl.projects.get(PROJECT_ID)
+
+            is_anonymous = username == default_user
+
+            variables = {
+                "TARGET_COMPONENTS_URL": repo_url,
+                "TARGET_COMPONENTS": " ".join(is_anonymous and default_user_target_components or target_components),
```

Comment:
So we no longer look at CI variables? That's kinda sad, sometimes I used to run the CI by hand with a custom value for `TARGET_COMPONENTS`. Now to do the same I need to log on the server, edit the config and restart the daemon.
Is that right?

Reply:
If you run it by hand on the gitlab UI it uses your value. This overrides the default when triggered by the CI hook.